sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/_polling.py

```diff
@@ -0,0 +1,102 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import logging
+import time
+
+from azure.core.polling import PollingMethod
+from azure.core.exceptions import ResourceNotFoundError, HttpResponseError
+
+logger = logging.getLogger(__name__)
+
+
+class DeleteResourcePoller(PollingMethod):
+    def __init__(self, interval=2):
+        self._command = None
+        self._deleted_resource = None
+        self._polling_interval = interval
+        self._status = "deleting"
+
+    def _update_status(self):
+        # type: () -> None
+        try:
+            self._command()
+            self._status = "deleted"
+        except ResourceNotFoundError:
+            self._status = "deleting"
+        except HttpResponseError as e:
+            if e.status_code != 403:
+                raise
```

Comment:
If the status is 403 we should set self._status to deleted

Reply:
switched, thanks!

Reply:
switched, thanks!